### ISSUE TITLE
LinkNN appears twice in the HDF5 Group and Dataset structure

### ISSUE THREAD

**Issue by bieryAtFnal** [DUNE-DAQ/dfmodules#8]
Adam,
I must have missed this when I reviewed the NestedGroups branch, but I'm now seeing the Link number as both an HDF5 Group and an HDF5 Dataset, as shown here:

```
   GROUP "TriggerRecord00008" {
      GROUP "FELIX" {
         GROUP "APA001" {
            GROUP "Link01" {
               DATASET "Link01" {
                  DATATYPE  H5T_STD_I8LE
                  DATASPACE  SIMPLE { ( 84, 1 ) / ( 84, 1 ) }
               }
            }
            GROUP "Link02" {
               DATASET "Link02" {
                  DATATYPE  H5T_STD_I8LE
                  DATASPACE  SIMPLE { ( 84, 1 ) / ( 84, 1 ) }
               }
            }
            GROUP "Link03" {
               DATASET "Link03" {
                  DATATYPE  H5T_STD_I8LE
                  DATASPACE  SIMPLE { ( 84, 1 ) / ( 84, 1 ) }
               }
            }
         }
      }
   }

```

Of course, it would be great if we could go back to just having a Link Dataset.

**Comment by adam-abed-abud** [comment]
Hi Kurt,

The bug is located in the "getSubGroup" method of HDF5FileUtils. The problem arises when looping on all the HDF5 groups of the given path and I was not considering that the last element is indeed the dataset. That is why in the output file you see both an HDF5 group with the same name of the HDF5 dataset. 

I have a fix for this bug in committed in branch `Store_TRH`.
Adam

**Comment by bieryAtFnal** [comment]
This issue was fixed for minidaqapp v1.0, so I'm going to close this Issue.
